app/controllers/reports_controller.rb

```diff
@@ -2,6 +2,54 @@ class ReportsController < ApplicationController
 
   before_action :authenticate_user!
 
+  def new
+    set_form_data
+    @structured_report_data = {}
+
+    @clients = Client.all
+    @projects = Project.all
+    @tasks = Task.all
+    @users = User.all
+  end
+
+  def update
+    set_form_data
+
+    scope_client_ids = @client_ids.empty? ? Client.ids : @client_ids
```

Comment:
May i suggest a slightly more elegant way to write this:
```suggestion
    scope_client_ids = @client_ids.presence || Client.ids
```